plugin/manager_linux.go

```diff
@@ -199,9 +199,25 @@ func (pm *Manager) upgradePlugin(p *v2.Plugin, configDigest digest.Digest, blobs
 
 	pdir := filepath.Join(pm.config.Root, p.PluginObj.ID)
 	orig := filepath.Join(pdir, "rootfs")
+
+	// Make sure nothing is mounted
+	// This could happen if the plugin was disabled with `-f` with active mounts.
+	// If the anything in `orig` is still mounted, this should error out.
+	mounts, err := getMounts(orig)
+	if err != nil {
+		return err
+	}
+	for _, m := range mounts {
+		if err := mount.Unmount(m); err != nil {
+			if mounted, e := mount.Mounted(m); e != nil || mounted {
+				return errors.Wrap(err, "error unmounting plugin root during upgrade")
```

Comment:
Looks like `recursiveUnmount` can be used. Also, not related to this PR, but https://github.com/cpuguy83/docker/blob/564839a8d8958ecc06600c8f730955c89cc32626/plugin/backend_linux.go#L651 error message should not include `while performing recursive unmount`